skbio/metadata/_testing.py

```diff
@@ -904,3 +905,245 @@ def test_has_positional_metadata(self):
+
+    def test_constructor_makes_shallow_copy_of_interval_metadata(self):
+        intvl = self.im.add(**self.intvls[1])
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+
+        self.assertEqual(obj.interval_metadata, self.im)
+        self.assertIsNot(obj.interval_metadata, self.im)
+
+        # Changing mutable value of metadata of the old interval
+        # also changes obj.
+        intvl.metadata['gene'].append('b')
+        self.assertEqual(obj.interval_metadata, self.im)
+
+        # Changing old interval doesn't change obj
+        intvl.locations = [(3, 6)]
+        self.assertNotEqual(obj.interval_metadata, self.im)
+
+    def test_eq_basic(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        im2 = IntervalMetadata(self.upper_bound)
+        im2.add(**self.intvls[0])
+        obj2 = self._interval_metadata_constructor_(self.upper_bound, im2)
+
+        self.assertReallyEqual(obj1, obj2)
+
+    def test_eq_populated_differently(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        obj2 = self._interval_metadata_constructor_(self.upper_bound)
+        obj2.interval_metadata.add(**self.intvls[0])
+
+        self.assertReallyEqual(obj1, obj2)
+
+    def test_eq_both_empty(self):
+        obj1 = self._interval_metadata_constructor_(self.upper_bound)
+        obj2 = self._interval_metadata_constructor_(self.upper_bound)
+        self.assertReallyEqual(obj1, obj2)
+
+    def test_ne_diff_len(self):
+        obj1 = self._interval_metadata_constructor_(0)
+        obj2 = self._interval_metadata_constructor_(self.upper_bound)
+        self.assertReallyNotEqual(obj1, obj2)
+
+    def test_ne_only_one_is_empty(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        obj2 = self._interval_metadata_constructor_(self.upper_bound)
+
+        self.assertReallyNotEqual(obj1, obj2)
+
+    def test_ne(self):
+        im1 = IntervalMetadata(self.upper_bound)
+        im1.add(**self.intvls[0])
+        obj1 = self._interval_metadata_constructor_(self.upper_bound, im1)
+
+        im2 = IntervalMetadata(self.upper_bound)
+        im2.add(**self.intvls[1])
+        obj2 = self._interval_metadata_constructor_(self.upper_bound, im2)
+
+        self.assertReallyNotEqual(obj1, obj2)
+
+    def test_copy_interval_metadata_empty(self):
+        obj = self._interval_metadata_constructor_(self.upper_bound, self.im)
+        obj_copy = copy.copy(obj)
+
+        self.assertEqual(obj, obj_copy)
+        self.assertIsNot(obj, obj_copy)
+
+    def test_copy_interval_metadata(self):
+        self.im.add(**self.intvls[0])
```

Comment:
can you use `self.intvls[1]` instead so that the mutable list value in `.metadata` can be asserted to be the same object? that's where the behavior differs between shallow vs. deep copy